pandas/core/indexes/base.py

```diff
@@ -450,7 +451,13 @@ def __new__(
                             return PeriodIndex(subarr, name=name, **kwargs)
                         except IncompatibleFrequency:
                             pass
-            return cls._simple_new(subarr, name)
+            if kwargs:
+                if len(kwargs) == 1:
+                    msg = f"Unexpected keyword argument {list(kwargs)[0]!r}"
+                else:
+                    msg = f"Unexpected keyword arguments {set(kwargs)!r}"
```

Comment:
I would just use this one (the set), no need to make a super-detailed error message